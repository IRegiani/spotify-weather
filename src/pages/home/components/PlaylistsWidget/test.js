import React from 'react';
import { shallow } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import Playlist, { fetchData, skeletonCard, generateListItem } from './index';

jest.mock('react-router-dom');

describe('Playlist Widget', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<Playlist weatherInfo={{ description: 'description' }} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test('renders loading skeletons correctly', () => {
    expect(skeletonCard).toMatchSnapshot();
  });

  test.skip('renders list item correctly', () => {
    const Component = generateListItem()();
    const wrapper = shallow(<Component />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test.todo('fetchData func');
});
