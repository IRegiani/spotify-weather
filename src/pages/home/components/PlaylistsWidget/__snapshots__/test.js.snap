// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Widget renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
  variant="outlined"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Skeleton))
      animation="wave"
      style={
        Object {
          "margin": "auto",
        }
      }
      variant="text"
      width={150}
    />
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))
        alignItems="flex-start"
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={40}
          variant="circle"
          width={40}
        />
        <div
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
        </div>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(ListItem))
        alignItems="flex-start"
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={40}
          variant="circle"
          width={40}
        />
        <div
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
        </div>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(ListItem))
        alignItems="flex-start"
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={40}
          variant="circle"
          width={40}
        />
        <div
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
        </div>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider)) />
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`Playlist Widget renders loading skeletons correctly 1`] = `
<React.Fragment>
  <WithStyles(ForwardRef(Skeleton))
    animation="wave"
    style={
      Object {
        "margin": "auto",
      }
    }
    variant="text"
    width={150}
  />
  <WithStyles(ForwardRef(List))>
    <React.Fragment>
      <WithStyles(ForwardRef(ListItem))
        alignItems="flex-start"
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={40}
          variant="circle"
          width={40}
        />
        <div
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
        </div>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider)) />
    </React.Fragment>
    <React.Fragment>
      <WithStyles(ForwardRef(ListItem))
        alignItems="flex-start"
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={40}
          variant="circle"
          width={40}
        />
        <div
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
        </div>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider)) />
    </React.Fragment>
    <React.Fragment>
      <WithStyles(ForwardRef(ListItem))
        alignItems="flex-start"
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={40}
          variant="circle"
          width={40}
        />
        <div
          style={
            Object {
              "marginLeft": "15px",
            }
          }
        >
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
          <WithStyles(ForwardRef(Skeleton))
            animation="wave"
            variant="text"
            width={350}
          />
        </div>
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(Divider)) />
    </React.Fragment>
  </WithStyles(ForwardRef(List))>
</React.Fragment>
`;
